hls-test-utils/src/Development/IDE/Test.hs

```diff
@@ -114,25 +114,25 @@ flushMessages = do
 --
 --   Rather than trying to assert the absence of diagnostics, introduce an
 --   expected diagnostic (e.g. a redundant import) and assert the singleton diagnostic.
-expectDiagnostics :: HasCallStack => [(FilePath, [(DiagnosticSeverity, Cursor, T.Text)])] -> Session ()
+expectDiagnostics :: HasCallStack => [(FilePath, [(DiagnosticSeverity, Cursor, T.Text, Maybe T.Text)])] -> Session ()
```

Comment:
Probably a good place to add doc about what `Maybe T.Text` is.

Reply:
I think I'll just make this four-tuple into its own data type. I'll make another for the expected diagnostic with a tag.

Reply:
Nooo... That way all the tests need to be rewritten. Ugh...

Reply:
I guess a type synonym it is then.